### PR TITLE
drivers/usbdev_synopsys_dwc2: fix and reenable DMA mode

### LINKED ISSUE #19359: sys/usbus: usbus_cdc_ecm does not work together with usbus_cdc_acm for STM32
#### Description

The combination of USBUS CDC ECM and CDC ACM interface does not work for different STM32 MCUs. It doesn't work for:
- STM32F429ZI FS interface (CID 1100)
- STM32F439ZI FS interface (CID 1200)
- STM32F723IE HS interface (CID 3100)
- STM32F746NG HS interface (CID 2100)

For these MCUs, the enumeration of the CDC ECM interface fails:
```
[224862.292833] usb 1-2.2: Product: nucleo-f439zi
[224862.292835] usb 1-2.2: Manufacturer: RIOT-os.org
[224862.292836] usb 1-2.2: SerialNumber: DC4DB8B40086FC7A
[224862.305285] cdc_acm 1-2.2:1.0: ttyACM1: USB ACM device
[224867.487946] cdc_ether: probe of 1-2.2:1.2 failed with error -22
...
[225333.844310] usb 1-4.1: Product: stm32f723e-disco
[225333.844311] usb 1-4.1: Manufacturer: RIOT-os.org
[225333.844313] usb 1-4.1: SerialNumber: A6BAC4E1B1E0806B
[225333.847348] cdc_acm 1-4.1:1.0: ttyACM1: USB ACM device
[225339.028774] cdc_ether: probe of 1-4.1:1.2 failed with error -32
```
It works for:
- STM32F723IE FS interface (CID 3000)
- STM32F746NG FS interface (CID 2000)
- STM32F767ZI FS interface (CID 2000)

For the FS interfaces it could be related to the 1.x Core version while the problem for the HS interfaces could be HS interface driver. `tests/usbus_cdc_ecm` works for all MCUs (for the HS interface PR #19358 is required).

The problem was already given before the `cpu/stm32/periph/usbdev_otg.c` was moved as modified version to `drivers/usbdev_synopsys_dwc2` (see branch [2022.07](https://github.com/RIOT-OS/RIOT/tree/2022.07-branch)).

#### Steps to reproduce the issue

1. Checkout PR #19358 
2. Use any of the following commands
   ```
    USEMODULE=stdio_cdc_acm BOARD=stm32f429i-disc1 make -j8 -C tests/usbus_cdc_ecm flash
    USEMODULE=stdio_cdc_acm BOARD=nucleo-f439zi make -j8 -C tests/usbus_cdc_ecm flash
    USEMODULE='periph_usbdev_hs_ulpi stdio_cdc_acm' BOARD=stm32f746g-disco make -j8 -C tests/usbus_cdc_ecm/ flash
    USEMODULE='periph_usbdev_hs_utmi stdio_cdc_acm' BOARD=stm32f723e-disco make -j8 -C tests/usbus_cdc_ecm flash
   ```

### PR DESCRIPTION
**Pull request by gschorcht**
### Contribution description

This PR fixes the DMA mode for all STM32 USB OTG HS cores (including that for STM32F4xx CID 1.xxx) and reenables it. It fixes remaining problems in issue #19359.

This PR includes also includes some changes that are needed to use the DMA mode:
- EP number is used as defined in CMSIS (if defined) for STM32
- `periph_usbdev_hs` feature is added in Kconfig
- `periph_usbdev_hs` feature is added in board definition of `stm32f429i-disc1`
- largest number of available EPs is used for STM32 instead of the smallest number (to be able to use all EPs of HS peripheral)
- `stm32f429i-disco` is removed from blacklist in `tests/usbus_cdc_ecm` since it uses the HS peripheral

### Testing procedure

The following tests should work
```python
USEMODULE=stdio_cdc_acm BOARD=stm32f429i-disc1 make -j8 -C tests/usbus_cdc_ecm flash
```
<details>
<summary>Test results</summary>

```python
[526755.875691] usb 1-2.2: new full-speed USB device number 106 using xhci_hcd
[526755.977853] usb 1-2.2: config 1 interface 3 altsetting 1 endpoint 0x84 has invalid maxpacket 512, setting to 64
[526755.977856] usb 1-2.2: config 1 interface 3 altsetting 1 endpoint 0x2 has invalid maxpacket 512, setting to 64
[526755.978762] usb 1-2.2: New USB device found, idVendor=1209, idProduct=7d01, bcdDevice= 1.00
[526755.978764] usb 1-2.2: New USB device strings: Mfr=3, Product=2, SerialNumber=4
[526755.978766] usb 1-2.2: Product: stm32f429i-disc1
[526755.978768] usb 1-2.2: Manufacturer: RIOT-os.org
[526755.978769] usb 1-2.2: SerialNumber: 7C156425A950A8EB
[526755.991190] cdc_acm 1-2.2:1.0: ttyACM1: USB ACM device
[526755.998131] cdc_ether 1-2.2:1.2 usb0: register 'cdc_ether' at usb-0000:00:14.0-2.2, CDC Ethernet Device, a6:f6:4a:85:1d:c9
[526756.044150] cdc_ether 1-2.2:1.2 enp0s20f0u2u2i2: renamed from usb0
```

</details>

```python
USEMODULE='stdio_cdc_acm periph_usbdev_hs_utmi' BOARD=stm32f723e-disco make -j8 -C tests/usbus_cdc_ecm flash
```
<details>
<summary>Test results</summary>

```python
[528733.480207] usb 1-4.3.4: reset high-speed USB device number 32 using xhci_hcd
[528733.707800] usb 1-4.4: new high-speed USB device number 111 using xhci_hcd
[528733.808257] usb 1-4.4: config 1 interface 0 altsetting 0 endpoint 0x81 has an invalid bInterval 255, changing to 11
[528733.808260] usb 1-4.4: config 1 interface 1 altsetting 0 bulk endpoint 0x1 has invalid maxpacket 64
[528733.808263] usb 1-4.4: config 1 interface 1 altsetting 0 bulk endpoint 0x82 has invalid maxpacket 64
[528733.808642] usb 1-4.4: New USB device found, idVendor=1209, idProduct=7d01, bcdDevice= 1.00
[528733.808645] usb 1-4.4: New USB device strings: Mfr=3, Product=2, SerialNumber=4
[528733.808647] usb 1-4.4: Product: stm32f723e-disco
[528733.808649] usb 1-4.4: Manufacturer: RIOT-os.org
[528733.808651] usb 1-4.4: SerialNumber: A6BAC4E1B1E0806B
[528733.811988] cdc_acm 1-4.4:1.0: ttyACM1: USB ACM device
[528733.814456] cdc_ether 1-4.4:1.2 usb0: register 'cdc_ether' at usb-0000:00:14.0-4.4, CDC Ethernet Device, e6:75:97:3a:74:ba
[528733.854371] cdc_ether 1-4.4:1.2 enp0s20f0u4u4i2: renamed from usb0
```

</details>

```python
USEMODULE='stdio_cdc_acm periph_usbdev_hs_ulpi' BOARD=stm32f746g-disco make -j8 -C tests/usbus_cdc_ecm flash
```
<details>
<summary>Test results</summary>

```python
[529000.944482] usb 1-4.3.4: reset high-speed USB device number 32 using xhci_hcd
[529003.728260] usb 1-4.4: new high-speed USB device number 114 using xhci_hcd
[529003.833107] usb 1-4.4: config 1 interface 0 altsetting 0 endpoint 0x81 has an invalid bInterval 255, changing to 11
[529003.833111] usb 1-4.4: config 1 interface 1 altsetting 0 bulk endpoint 0x1 has invalid maxpacket 64
[529003.833113] usb 1-4.4: config 1 interface 1 altsetting 0 bulk endpoint 0x82 has invalid maxpacket 64
[529003.833743] usb 1-4.4: New USB device found, idVendor=1209, idProduct=7d00, bcdDevice= 1.00
[529003.833747] usb 1-4.4: New USB device strings: Mfr=3, Product=2, SerialNumber=4
[529003.833749] usb 1-4.4: Product: stm32f746g-disco
[529003.833751] usb 1-4.4: Manufacturer: RIOT-os.org
[529003.833753] usb 1-4.4: SerialNumber: 66FE8934D1A363E0
[529003.837143] cdc_acm 1-4.4:1.0: ttyACM1: USB ACM device
[529003.839755] cdc_ether 1-4.4:1.2 usb0: register 'cdc_ether' at usb-0000:00:14.0-4.4, CDC Ethernet Device, 6a:88:1f:1f:b1:f0
[529003.879025] cdc_ether 1-4.4:1.2 enp0s20f0u4u4i2: renamed from usb0```
```
</details>

### Issues/PRs references

Fixes #19359

### CODE REVIEW

**Comment by dylad** [comment]
I can have a look at it next week.

File: cpu/stm32/include/periph_cpu.h [hunk 1]
```diff
@@ -131,33 +131,40 @@ typedef struct {
...
 #endif
 
 /**
  * @brief Number of IN/OUT endpoints including EP0 as used by USBUS
  *
- * @note Since only a single number of EPs can be defined for USBUS that is
- *       valid for all devices, the smallest number of EPs must be used for
- *       multiple USB devices.
+ * @note USBUS allows only one definition of the number of available EPs, which
+ *       is then used for all devices. To be able to use all EPs for devices
```
[the remark this answers is not in the document]
Comment (gschorcht) [reply cpu/stm32/include/periph_cpu.h:158]: `device` means in this context one USB device instance of type `usbdev_t`.

**Comment by MrKevinWeiss** [comment]
> Since this fixes a regression, I think it should be in the release.

Fixes are always welcome!

**Comment by benpicco** [comment]
bors merge

**Comment by gschorcht** [comment]
Hm, indeed. Good catch. I suspect that backporting PR #19472 makes no sense, or at least has no effect without backporting that PR, if it doesn't actually impact the function.

So yes, we should also backport this one.

**Comment by MrKevinWeiss** [comment]
I think this is actually in already in I think... my bad.

**Pull request merged**